MantidPlot/src/Mantid/MantidMatrix.cpp

```diff
@@ -596,8 +596,8 @@ QwtDoubleRect MantidMatrix::boundingRect() {
       bool theSame = true;
       double dx = 0.;
       for (int i = i0; i <= m_endRow; ++i) {
-        if (m_workspace->readX(i).front() != x_start ||
-            m_workspace->readX(i).back() != x_end) {
+        if (m_workspace->x(i).front() != x_start ||
```

Comment:
Ditto about asking for `x(i)` twice - instead take a const reference and reuse that?